### PR TITLE
[web-components] refactor color recipes for DI

### PR DESCRIPTION
**Pull request by chrisdholt**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes
Creates vNext color recipes and updates tests to ensure parity.

#### Focus areas to test

(optional)

### CODE REVIEW

**Review by nicholasrice — APPROVED** [review]

File: packages/web-components/src/color-vNext/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# FAST Color Recipes
```
Comment (EisenbergEffect) [inline packages/web-components/src/color-vNext/README.md:1]:
```suggestion
# FluentUI Color Recipes
```

File: packages/web-components/src/color-vNext/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+Color recipes are named colors who's value is algorithmically defined from a variety of inputs. `@microsoft/fast-components` relies on these recipes heavily to achieve expressive theming options while maintaining color accessability targets.
```
Comment (EisenbergEffect) [inline packages/web-components/src/color-vNext/README.md:3]:
```suggestion
Color recipes are named colors who's value is algorithmically defined from a variety of inputs. `@fluentui/web-components` relies on these recipes heavily to achieve expressive theming options while maintaining color accessibility targets.
```

File: packages/web-components/src/color-vNext/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+A Swatch is a representation of a color that has a `relativeLuminance` value and a method to convert the swatch to a color string. It is used by recipes to determine which colors to use for UI.
+
+### SwatchRGB
+
+A concrete implementation of `Swatch`, it is a swatch with red, green, and blue 64bit color channels .
+
+**Example: Creating a SwatchRGB**
+
+```ts
+import { SwatchRGB } from '@microsoft/fast-components';
```
Comment (EisenbergEffect) [inline packages/web-components/src/color-vNext/README.md:16]:
```suggestion
import { SwatchRGB } from '@fluentui/web-components';
```

File: packages/web-components/src/color-vNext/palette.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
+import { clamp, ColorRGBA64, ComponentStateColorPalette, parseColorHexRGB } from '@microsoft/fast-colors';
```
Comment (EisenbergEffect) [inline packages/web-components/src/color-vNext/palette.ts:1]: I wonder if general purpose types like `Palette` should actually be in foundation instead?

**Review by EisenbergEffect — APPROVED** [review]
My only comments are that there's a large amount of what appears to be duplicative code with fast-components. Should any of this be moved into foundation? Some of it seems like primitives e.g. Palette. Also, what was our reason for making the recipes internal? Is that a temporary thing that we plan to lift after we work out a few more things?

File: packages/web-components/src/color-vNext/palette.ts (hunk 4 above)
Comment (chrisdholt) [reply packages/web-components/src/color-vNext/palette.ts:1]: Right now I'm not sure that these won't potentially change or diverge - I do think we can create a corresponding issue in FAST to track potentially breaking some of the color work into it's own package, but I'd prefer a more robust data set before we do that.

**Pull request merged**